src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -564,10 +564,10 @@ class CLISecurityDomainOperation(str, Enum):
     # endregion
 
     # region keyvault security-domain
-    for scope in ['init-recovery', 'download', 'upload']:
+    for scope in ['init-recovery', 'download', 'upload', 'wait']:
         with self.argument_context('keyvault security-domain {}'.format(scope), arg_group='HSM Id') as c:
-            c.argument('hsm_name', hsm_url_type, required=False,
-                       help='Name of the HSM. Can be omitted if --id is specified.')
+            c.extra('hsm_name', hsm_url_type, required=False,
+                    help='Name of the HSM. Can be omitted if --id is specified.')
```

Comment:
If `hsm_name` is now an "extra" argument, how can it be used to create the SDK client?

Reply:
It's now accepted when creating client in `client_factory.py` instead of calling sdk in `custom.py`